src/main/java/dev/j3fftw/litexpansion/Events.java

```diff
@@ -161,101 +157,101 @@ public void onDye(PlayerInteractEntityEvent e) {
+        final Block block = event.getClickedBlock();
         if (block == null) {
             return;
         }
 
         final Material blockType = block.getType();
         final Location blockLocation = block.getLocation();
+        final ItemStack hand = event.getItem();
 
         final MiningDrill miningDrill = (MiningDrill) SlimefunItem.getById(Items.MINING_DRILL.getItemId());
+        if (miningDrill.isItem(hand)) {
+            Validate.notNull(miningDrill);
+            if (!Check(miningDrill, event, blockLocation)) {
+                return;
+            }
 
-        Validate.notNull(miningDrill, "Can not be null");
-        if (miningDrill.isItem(e.getItem())
-            && SlimefunTag.STONE_VARIANTS.isTagged(blockType)
-            && !miningDrill.isDisabled()
-            && Slimefun.getProtectionManager().hasPermission(e.getPlayer(),
-            blockLocation, Interaction.BREAK_BLOCK)
-        ) {
-            e.setCancelled(true);
+            if (!SlimefunTag.STONE_VARIANTS.isTagged(blockType)) {
+                return;
+            }
 
-            final SlimefunItem slimefunItem = BlockStorage.check(block);
+            DrillUse(0.5f, block, blockType, blockLocation, event);
+        }
 
-            if (slimefunItem == null && ((Rechargeable) SlimefunItem.getByItem(e.getItem()))
-                .removeItemCharge(e.getItem(), 0.5F)
-            ) {
-                // This allows other plugins to register broken block as player broken
-                BlockBreakEvent newEvent = new BlockBreakEvent(block, e.getPlayer());
-                Bukkit.getServer().getPluginManager().callEvent(newEvent);
-                block.setType(Material.AIR);
-                e.getPlayer().playSound(blockLocation, Sound.BLOCK_STONE_BREAK, 1.5F, 1F);
-
-                // This has to be done because the item in the main hand is not a pickaxe
-                if (blockType == Material.STONE) {
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(Material.COBBLESTONE)
-                    );
-
-                } else {
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(blockType)
-                    );
-                }
+        final MiningDrill diamondDrill = (MiningDrill) SlimefunItem.getById(Items.DIAMOND_DRILL.getItemId());
+        if (diamondDrill.isItem(hand)) {
+
+            Validate.notNull(diamondDrill);
+            if (!Check(diamondDrill, event, blockLocation)) {
+                return;
             }
+
+            if (!SlimefunTag.MINEABLE_PICKAXE.isTagged(blockType)) {
+                return;
+            }
+
+            DrillUse(1.5f, block, blockType, blockLocation, event);
         }
     }
+    
+    public boolean Check(MiningDrill miningDrill, PlayerInteractEvent event, Location blockLocation) {
+        return miningDrill.isItem(event.getItem())
```

Comment:
So, I may be wrong here - but you have covered isItem() before you get here, hasPermission() is covered by the event.isCancelled (I think, check this!). So I think (think!) this method can be replaced with an in-line isDisabled (which if you moved this whole thing into an ItemHandler you wouldnt't even need that - but thats a whoooole other conversation.

Sorry if im wrong on this - im really just glancing inbetween cooking dinner bits xD

Reply:
i tested and this isnt the case